src/Orc.DataAccess/Database/Extensions/StringExtensions.cs

```diff
@@ -8,19 +8,25 @@ public static class StringExtensions
     {
         public static string EncryptConnectionString(this string connectionString, string providerName)
         {
+            ArgumentNullException.ThrowIfNull(connectionString);
+            Argument.IsNotNullOrEmpty(() => providerName);
+
             return AlterConnectionStringPropertyValue(connectionString, providerName, x => x.Encrypt());
         }
 
         public static string DecryptConnectionString(this string connectionString, string providerName)
         {
+            ArgumentNullException.ThrowIfNull(connectionString);
+            Argument.IsNotNullOrEmpty(() => providerName);
+
             return AlterConnectionStringPropertyValue(connectionString, providerName, x => x.Decrypt());
         }
 
-        private static string AlterConnectionStringPropertyValue(this string connectionString, string providerName, Func<string, string> alteractionFunction)
+        private static string AlterConnectionStringPropertyValue(this string connectionString, string providerName, Func<string, string?> alteractionFunction)
```

Comment:
I think it should be alterActionFunction or alterFunction, I don't think alteraction is a valid word.